scalar.c

```diff
@@ -232,27 +232,30 @@ static int set_recommended_config(int reconfigure)
 
 	/*
 	 * Set HTTP/1.1 for Azure DevOps URLs
+	 * We check for dev.azure.com and .visualstudio.com/ patterns
+	 * which are sufficient to identify ADO URLs
 	 */
 	if (!git_config_get_string("remote.origin.url", &value)) {
-		if (strstr(value, "dev.azure.com") || 
-		    strstr(value, "visualstudio.com")) {
+		if (starts_with(value, "https://dev.azure.com") || 
+		    strstr(value, ".visualstudio.com/")) {
 			struct strbuf key = STRBUF_INIT;
 			strbuf_addf(&key, "http.%s.version", value);
+			char *original_value = value;
```

Comment:
We do not need the original value anymore here. Let's use `FREE_AND_NULL(value);`.